packages/@lwc/ssr-compiler/src/__tests__/fixtures.spec.ts

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * Naive HTML fragment formatter.
+ *
+ * This is a replacement for Prettier HTML formatting. Prettier formatting is too aggressive for
+ * fixture testing. It not only indent the HTML code but also fixes HTML issues. For testing we want
+ * to make sure that the fixture file is as close as possible to what the engine produces.
+ */
+function formatHTML(src: string): string {
+    let res = '';
+    let pos = 0;
+    let start = pos;
+
+    let depth = 0;
+
+    const getPadding = () => {
+        return '  '.repeat(depth);
+    };
+
+    while (pos < src.length) {
+        // Consume element tags and comments.
+        if (src.charAt(pos) === '<') {
+            const tagNameMatch = src.slice(pos).match(/(\w+)/);
+
+            if (!tagNameMatch) {
+                throw new Error(
+                    `Expected to find tagname at pos ${pos} but found "${src.slice(pos, 20)}"`
+                );
+            }
+
+            // Special handling for `<style>` tags – these are not encoded, so we may hit '<' or '>'
+            // inside the text content. So we just serialize it as-is.
+            if (tagNameMatch[0] === 'style') {
+                const styleMatch = src.slice(pos).match(/<style([\s\S]*?)>([\s\S]*?)<\/style>/);
+                if (styleMatch) {
+                    // opening tag
+                    const [wholeMatch, attrs, textContent] = styleMatch;
+                    res += getPadding() + `<style${attrs}>` + '\n';
+                    depth++;
+                    res += getPadding() + textContent + '\n';
+                    depth--;
+                    res += getPadding() + '</style>' + '\n';
+                    start = pos = pos + wholeMatch.length;
+                    continue;
+                }
+            }
+
+            const isVoid = isVoidElement(tagNameMatch[0], HTML_NAMESPACE);
+            const isClosing = src.charAt(pos + 1) === '/';
+            const isComment = src.slice(pos, pos + 3) === '!--';
+
+            start = pos;
+            while (src.charAt(pos++) !== '>') {
+                // Keep advancing until consuming the closing tag.
+            }
+
+            // Adjust current depth and print the element tag or comment.
+            if (isClosing) {
+                depth--;
+            }
+
+            res += getPadding() + src.slice(start, pos) + '\n';
+
+            const isSelfClosing = src.charAt(pos - 2) === '/';
+            if (!isClosing && !isSelfClosing && !isVoid && !isComment) {
+                depth++;
+            }
+        }
+
+        // Consume text content.
+        start = pos;
+        while (src.charAt(pos) !== '<' && pos < src.length) {
+            pos++;
+        }
+
+        if (start !== pos) {
+            res += getPadding() + src.slice(start, pos) + '\n';
+        }
+    }
+
+    return res.trim();
+}
+
+function testFixtures() {
+    testFixtureDir(
+        {
+            root: path.resolve(__dirname, '../../../engine-server/src/__tests__/fixtures'),
```

Comment:
I feel there should be a cleaner way to do this. We should be able to centralize all the fixtures in one place.

Maybe we could have an internal-only package that has both `@lwc/engine-server` and `@lwc/ssr-compiler` as dependencies and is only responsible for running the fixture tests.

Reply:
Yes, I was saying that on the call with folks that'll be working on this. I absolutely agree - this should be moved out into a common location. I'll add a follow-up work item for this.